script.skin.helper.service/resources/lib/main_module.py

```diff
@@ -0,0 +1,695 @@
+                listitem.setProperty("path", media["file"])
+                results.append(listitem)
+
+        # finished lookup - display listing with results
+        xbmc.executebuiltin("dialog.Close(busydialog)")
+        dialog = DialogSelect("DialogSelect.xml", "", listing=results, windowtitle=window_header,
+                              multiselect=False, richlayout=True)
+        dialog.doModal()
+        result = dialog.result
+        del dialog
+        if result:
+            if xbmc.getCondVisibility(
+                    "Window.IsActive(script-skin_helper_service-CustomInfo.xml) | "
+                    "Window.IsActive(movieinformation)"):
+                xbmc.executebuiltin("Dialog.Close(movieinformation)")
+                xbmc.executebuiltin("Dialog.Close(script-skin_helper_service-CustomInfo.xml)")
+                xbmc.sleep(1000)
+            xbmc.executebuiltin('PlayMedia("%s")' % result.getProperty("path"))
+            del result
+
+    def getcastmedia(self):
+        '''helper to show a dialog with all media for a specific actor'''
+        xbmc.executebuiltin("ActivateWindow(busydialog)")
+        name = self.params.get("name", "")
+        window_header = self.params.get("name", "")
+        results = []
+        items = self.kodidb.castmedia(name)
+        items = process_method_on_list(self.kodidb.prepare_listitem, items)
+        for item in items:
+            if item["file"].startswith("videodb://"):
+                item["file"] = "ActivateWindow(Videos,%s,return)" % item["file"]
+            else:
+                item["file"] = 'PlayMedia("%s")' % item["file"]
+            results.append(self.kodidb.create_listitem(item, False))
+        # finished lookup - display listing with results
+        xbmc.executebuiltin("dialog.Close(busydialog)")
+        dialog = DialogSelect("DialogSelect.xml", "", listing=results, windowtitle=window_header, richlayout=True)
+        dialog.doModal()
+        result = dialog.result
+        del dialog
+        if result:
+            while xbmc.getCondVisibility("System.HasModalDialog"):
+                xbmc.executebuiltin("Action(Back)")
+                xbmc.sleep(300)
+            xbmc.executebuiltin(result.getfilename())
+            del result
+
+    def setfocus(self):
+        '''helper to set focus on a list or control'''
+        control = self.params.get("control")
+        fallback = self.params.get("fallback")
+        position = self.params.get("position", "0")
+        relativeposition = self.params.get("relativeposition")
+        if relativeposition:
+            position = int(relativeposition) - 1
+        count = 0
+        if control:
+            while not xbmc.getCondVisibility("Control.HasFocus(%s)" % control):
+                if xbmc.getCondVisibility("Window.IsActive(busydialog)"):
+                    continue
+                elif count == 20 or (xbmc.getCondVisibility(
```

Comment:
count will never reach 20 i think?
it's increased by 1 and you break from the loop.